### ISSUE TITLE
Use Git LFS to manage binary appendix of this repo

### ISSUE THREAD

**Issue by virtualritz** [johannesvollmer/exrs#87]
This repo is 718 MB.
Can we move the examples & co into a [Git LFS](https://git-lfs.github.com/) managed portion so when cloning they are optional?

It's not only ergonomics but also about inclusiveness.
There are devs in third world countries that may want to clone this repo and they simply can't or will have to go to lots of pain to do so.

How do I know? I just tried to clone this repo from my parent's house in the countryside with a 256k connection. Go figure. :)

**Comment by johannesvollmer** [comment]
> move the examples & co into a Git LFS managed portion 

What exactly are the steps for this? I'm not that experienced with this topic

**Comment by johannesvollmer** [comment]
> Can we move the examples & co 

Actually, the purpose of the images is automated testing, which is why you would still need to download them as soon as you want to run the tests. 

I checked the link and it says that we can just make it track all exrs files in the repo, which sounds really simple. I can try this right now

**Comment by virtualritz** [comment]
Yes, of course. The idea is that you can grab the repo w/o this.
I also found Git LFS to be more stable downloading such large files over spotty connections.

**Comment by johannesvollmer** [comment]
![image](https://user-images.githubusercontent.com/32042925/88214475-9066ca80-cc5a-11ea-9a1d-ace212512e95.png)

Some fuzzed EXR files from the official repository are not marked as `.exr` file in order to prevent people from accidentially opening them. We will have to define a special rule for them later, I guess

**Comment by johannesvollmer** [comment]
Following the suggestions from the Git LFS website results in the existing gitignore file being replaced, which results in thousands of temporary compilation files and IDE settings to be commitet.... this is not what I wanted, LFS :(

**Comment by virtualritz** [comment]
That sucks. I never converted a repo. I only set up new repos with LFS.
I presume you also found [_this_](https://support.atlassian.com/bitbucket-cloud/docs/use-bfg-to-migrate-a-repo-to-git-lfs/) via google and it doesn't help?

**Comment by johannesvollmer** [comment]
Yeah, just reverting all those 5,000 file changes, takes a while lol

**Comment by johannesvollmer** [comment]
Wow ok... Git LFS rewrote the entire history so I cannot simply undo these changes wow... (`--no-rewrite` did not work thanks LFS)

**Comment by virtualritz** [comment]
That's why they tell you to backup the repo. :]
I have a pristine copy from today, in case anything goes wrong.

**Comment by johannesvollmer** [comment]
Nice :D I have never pushed the local repository so nothing should have any effect, now I have my local backup

**Comment by johannesvollmer** [comment]
Although I thought the whole point of Git was to never loose anything... makes no sense any more if you can still loose everything that easily

**Comment by johannesvollmer** [comment]
So this happened:
```
D:\Documents\Rust\exrs>git lfs migrate import --fixup --everything

migrate: Sorting commits: ..., done
migrate: Rewriting commits: 100% (578/578), done
  master                ad472beb19d5509654d7d6be3db6d49cfdcce7b6 -> ad472beb19d5509654d7d6be3db6d49cfdcce7b6
  project-structure     3ef417a1460244030c6bfec80351e8602725e339 -> 3ef417a1460244030c6bfec80351e8602725e339
  0.7.3                 e7e551090ced464a137a2aeb19464b271c02683b -> e7e551090ced464a137a2aeb19464b271c02683b
  v0.6.1                2b72e8d4281cd45f40835e8a4aec5f4e2e9d27ca -> 2b72e8d4281cd45f40835e8a4aec5f4e2e9d27ca
  v0.7.0                954b88c3534cb0560b70ea944835348c92e55923 -> 954b88c3534cb0560b70ea944835348c92e55923
  v0.7.3                cb1a83ac19ba9ef7d95afc674b974a369fe51824 -> cb1a83ac19ba9ef7d95afc674b974a369fe51824
  v0.7.4                cb1a83ac19ba9ef7d95afc674b974a369fe51824 -> cb1a83ac19ba9ef7d95afc674b974a369fe51824
  v0.8.0                bfdbf1f65dcc8dfa151ec26196bb7604ab79edc2 -> bfdbf1f65dcc8dfa151ec26196bb7604ab79edc2
migrate: Updating refs: ..., done
migrate: checkout: ..., done

```

**Comment by johannesvollmer** [comment]
Nothing changed though, how can I check that it even worked?

**Comment by johannesvollmer** [comment]
`git lfs ls-files` does not list any files

**Comment by virtualritz** [comment]
Maybe the binary files are now staged for commit? `git lfs ls-files` won't list files which are not committed yet.
Did you try `git lfs status`?

**Comment by johannesvollmer** [comment]
`git lfs status` tells me that the git attributes file is not staged:
```
Git LFS objects not staged for commit:

        .gitattributes (Git: 30750ee -> File: fb688a8)

```

It's not wrong, the file is not staged as it is already commited and pushed.... so do I need to undo that push now?

**Comment by johannesvollmer** [comment]
Changing the file and then doing that again yielded an empty LFS status and an empty LFS ls-files

**Comment by johannesvollmer** [comment]
I'm going to bed now and maybe I'll do this later... it's certainly not "just working" yet

**Comment by virtualritz** [comment]
Ah, it's amazing enough you tried immediately. I just created this issue so I don't forget. Didn't expect anyone to jump on it right away. Cheers for the effort.

**Comment by johannesvollmer** [comment]
Ok so, at what point does git LFS actually download the large files? Certainly not when you run `cargo test`, right? Do you have to do that manually? Or do you have to manually specify that you don't want to download the large files when you are cloning?

**Comment by dgsantana** [comment]
You must use `git lfs install` for any repo just cloned. I have been using
for years, but it's easy to forget that step if using command line.

Johannes Vollmer <notifications@github.com> escreveu no dia sábado,
25/07/2020 à(s) 11:26:

**Comment by johannesvollmer** [comment]
That makes sense, got to put that info the readme, thanks ^^

**Comment by johannesvollmer** [comment]
> I just tried to clone this repo from my parents’ house, in the countryside, with a 256k connection

So, as you probably have installed git LFS, this is probably not going to change things for you? I mean, you'd have to uninstall git LFS I guess?

**Comment by johannesvollmer** [comment]
@virtualritz Could you checkout PR #88, to check whether everything works? Maybe review it?

**Comment by johannesvollmer** [comment]
The mentioned PR requires a monthly fee and thus was discarded. This issue can be used to discuss other options for binary data, such as submodules.
